.github/workflows/snackpub.yml

```diff
@@ -0,0 +1,184 @@
+name: Snackpub
+
+defaults:
+  run:
+    working-directory: snackpub
+
+concurrency:
+  group: snackpub-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  workflow_dispatch:
+    inputs:
+      deploy:
+        description: What environment should be deployed
+        type: choice
+        default: no-deploy
+        options:
+          - no-deploy
+          - staging
+          - production
+  pull_request:
+    paths:
+      - .github/actions/setup-google-cloud/**
+      - .github/actions/setup-secrets/**
+      - .github/actions/setup-snackpub/**
+      - .github/workflows/snackpub.yml
+      - snackpub/**
+      - .eslint*
+      - .prettier*
+      - yarn.lock
+  push:
+    branches: [main]
+    paths:
+      - .github/actions/setup-google-cloud/**
+      - .github/actions/setup-secrets/**
+      - .github/actions/setup-snackpub/**
+      - .github/workflows/snackpub.yml
+      - snackpub/**
+      - .eslint*
+      - .prettier*
+      - yarn.lock
+
+jobs:
+  review:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 🏗 Setup repository
+        uses: actions/checkout@v3
+
+      - name: 🏗 Setup secrets
+        uses: ./.github/actions/setup-secrets
+        with:
+          git-crypt-key: ${{ secrets.GIT_CRYPT_KEY }}
+
+      - name: 🏗 Setup snackpub
+        uses: ./.github/actions/setup-snackpub
+
+      - name: 🚨 Lint snackpub
+        run: yarn lint --max-warnings 0
+
+  build:
+    if: ${{ github.event_name == 'pull_request' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: 🏗 Setup repository
+        uses: actions/checkout@v3
+
+      - name: 🏗 Setup secrets
+        uses: ./.github/actions/setup-secrets
+        with:
+          git-crypt-key: ${{ secrets.GIT_CRYPT_KEY }}
+
+      - name: 🏗 Setup Google Cloud SDK
+        uses: ./.github/actions/setup-google-cloud
+        with:
+          project-id: exponentjs
+          project-zone: us-central1
+          project-cluster: general-central
+          service-key: ${{ secrets.GCLOUD_KEY }}
+
+      - name: 🛠 Build snackpub
+        run: skaffold build --filename snackpub/skaffold.yaml --file-output snackpub/build.json
```

Comment:
There's no need to use the `--file-output` flag if the deploy script always uses the `latest` tag instead of the contents of `build.json`